testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,693 @@
+                    if (!fieldList[i].getName().equals(subFieldName)) {
+                        continue;
+                    }
+                    try {
+                        fieldList[i].setAccessible(true);
+                        Field modifiersField = Field.class.getDeclaredField("modifiers");
+                        modifiersField.setAccessible(true);
+                        modifiersField.setInt(fieldList[i], fieldList[i].getModifiers() & ~Modifier.FINAL);
+                    }
+                    catch (Exception e) {
+                        String outputPrivateError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputPrivateError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                    try {
+                        ob_0 = fieldList[i].get(null);
+                        boolean threadLocal = false;
+                        if (ob_0 instanceof ThreadLocal) {
+                            threadLocal = true;
+                        }
+                        try {
+                            // directly invoke reset when dealing with a Mockito mock object
+                            if (Mockito.mockingDetails(ob_0).isMock()) {
+                                Method m = Mockito.class.getDeclaredMethod("reset", Object[].class);
+                                m.invoke(null, new Object[]{new Object[]{ob_0}});
+                            } else {
+                                System.setProperty("currentClassInXStream", fieldList[i].getDeclaringClass().getName());
+                                System.setProperty("currentFieldInXStream", fieldList[i].getName());
+                                UnmarshalChain.reset();
+                                UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                                ob_0 = xstream.fromXML(state0);
+                            }
+                        } catch (NoSuchMethodError nsme) {
+                            System.setProperty("currentClassInXStream", fieldList[i].getDeclaringClass().getName());
+                            System.setProperty("currentFieldInXStream", fieldList[i].getName());
+                            UnmarshalChain.reset();
+                            UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                            ob_0 = xstream.fromXML(state0);
+                        }
+                        try {
+                            if (AccessibleObject.class.isAssignableFrom(ob_0.getClass())) {
+                                ((AccessibleObject) ob_0).setAccessible(true);
+                            }
+                        } catch (NullPointerException npex) {
+                            npex.printStackTrace();
+                        }
+                        if (threadLocal) {
```

Comment:
Should this logic concerning `threadLocal` be moved higher up or surrounding more code, since we don't necessarily want to do all the above deserialization if we're going to deserialize differently if `threadLocal` is true?